modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1298,15 +1308,15 @@ void ONNXImporter::parseReduce(LayerParams& layerParams, const opencv_onnx::Node
 
                 std::swap(perm[index], perm[i]);
                 std::swap(inpShape[index], inpShape[i]);
-                needPermuet = true;
+                needPermute = true;
```

Comment:
Blame: this implementation is not robust and causing problems when it comes to support other backends.

Potential problem: if Reduce is operated on multiple axes, for example [1, -1], permute makes it wrong.

Problems of different backend: In the CPU impl, permute -> reduce does something like this to the shape [0, 2, 3, 1] -> [0, 1, 2, 3]. A hidden permute is executed in the impl of reduce, which leads to the failure of shape inference for other backends.

We need to refactor the reduce importer and implementation.